packages/react-storage/src/components/StorageBrowser/context/controls/Navigate/__tests__/Navigate.spec.tsx

```diff
@@ -1,159 +1,106 @@
+  permission: location.permission,
+  scope: `s3://${location.bucket}/*`,
+  type: location.type,
+};
+
 describe('navigateReducer', () => {
-  const location: LocationData = {
-    bucket: 'bucket-name',
-    scope: 's3://',
-    permission: 'READ',
-    type: 'OBJECT',
-    prefix: undefined,
-  };
-
-  it('handles a SELECT_LOCATION as expected', () => {
+  it('ACCESS_LOCATION action sets the location and bucket as the initial entry value', () => {
     const initialState: NavigateState = {
+      history: [],
       location: undefined,
-      history: undefined,
-    };
-
-    const action: NavigateAction = { type: 'SELECT_LOCATION', location };
-
-    const expectedState: NavigateState = {
-      location: {
-        bucket: location.bucket,
-        permission: location.permission,
-        scope: `s3://${location.bucket}/*`,
-        type: location.type,
-        prefix: location.prefix,
-      },
-      history: undefined,
     };
 
-    const newState = navigateReducer(initialState, action);
-
-    expect(newState).toEqual(expectedState);
-  });
-
-  it('handles a DESELECT_LOCATION as expected', () => {
-    const state: NavigateState = {
-      location,
-      history: ['folder1/', 'folder2/'],
+    const action: NavigateAction = {
+      type: 'ACCESS_LOCATION',
+      location: locationAccess,
     };
 
-    const action: NavigateAction = { type: 'DESELECT_LOCATION' };
-
     const expectedState: NavigateState = {
-      location: undefined,
-      history: undefined,
+      history: [location.bucket],
+      location: locationAccess,
     };
 
-    const newState = navigateReducer(state, action);
+    const newState = navigateReducer(initialState, action);
 
-    expect(newState).toEqual(expectedState);
+    expect(newState).toStrictEqual(expectedState);
   });
 
-  it('handles a ENTER_FOLDER as expected', () => {
+  it('handles a NAVIGATE action as expected when target location is root', () => {
     const state: NavigateState = {
-      location,
-      history: ['folder1/'],
+      history: [location.bucket, location.prefix],
+      location: locationAccess,
     };
 
     const action: NavigateAction = {
-      type: 'ENTER_FOLDER',
-      name: 'folder2/',
+      type: 'NAVIGATE',
+      prefix: location.bucket,
     };
 
     const expectedState: NavigateState = {
-      location: {
-        bucket: location.bucket,
-        permission: location.permission,
-        scope: `s3://${location.bucket}/${action.name}*`,
-        type: location.type,
-        prefix: action.name,
-      },
-      history: ['folder1/', 'folder2/'],
+      history: [location.bucket],
+      location: locationAccess,
     };
 
     const newState = navigateReducer(state, action);
 
-    expect(newState).toEqual(expectedState);
+    expect(newState).toStrictEqual(expectedState);
   });
 
-  it('should handle ENTER_FOLDER with undefined history', () => {
+  it('handles a NAVIGATE action to a new location', () => {
```

Comment:
not blocking, but I think having another test navigating back from a prefix could be helpful too?


```jsx
  it('handles a NAVIGATE action to an old location', () => {
    const state: NavigateState = {
      history: [location.bucket, locationTwo.prefix, 'public/folder1/', 'public/folder1/nestedFolder/'],
      location: locationAccess,
    };
    
    const action: NavigateAction = {
      type: 'NAVIGATE',
      prefix: locationTwo.prefix,
    };
    
    const expectedState: NavigateState = {
      history: [location.bucket, locationTwo.prefix],
      location: locationAccess,
    };

    const newState = navigateReducer(state, action);

    expect(newState).toStrictEqual(expectedState);
  }
```